resources/scripts/non_spacing_marks.py

```diff
@@ -0,0 +1,58 @@
+"""glyphsLib uses GlyphData.xml to know things about glyphs.
+
+Naively you would think we could get the same information by querying unicode
+properties but it seems not, for example brevecomb-cy with no mapped codepoint
+can be identified as a non-spacing mark by name. No unicode db will do that for us.
+
+This utility generates a Rust file with key information extracted from GlyphData.xml.
+"""
+
+from absl import app
+from lxml import etree
+from pathlib import Path
+import sys
+
+
+def main(_):
+    glyphs_lib = Path("../glyphsLib")
+    if not glyphs_lib.is_dir():
+        sys.exit(f"Missing {glyphs_lib}")
+    glyphdata_file = glyphs_lib / "Lib/glyphsLib/data/GlyphData.xml"
```

Comment:
instead of expecting that glyphsLib source is cloned in a sibling folder, you could rely on the fact that these data files are shipped alongside the python modules as resources and use importlib to get to them, like glyphsLib itself does:

https://github.com/googlefonts/glyphsLib/blob/6372da0c1f77cad3152364c2473b7a380ac6c301/Lib/glyphsLib/glyphdata.py#L106-L108C7

this way, it doesn't matter where glyphsLib is, it's sufficient that it is importable. You could even get the `glyphsLib.__version__` and write that out to the generated .rs file so this is recorded (the glyphdata xml may change from time to time and it'd be good to track exactly which one we are using).

Reply:
Don't know if you noticed, for CJK glyphs there's an additional XML data file as well.
Actually, instead of parsing it yourself with etree, since you are already requiring glyphsLib to be present, why don't you simply import glyphsLib.glyphdata and let it load both the GlyphDatas for you. Then you'd query them for the attributes that you are interested in. E.g. something like this:

```python
from glyphsLib.glyphdata import GlyphData

with (
    open_binary("glyphsLib.data", "GlyphData.xml") as f1,
    open_binary("glyphsLib.data", "GlyphData_Ideographs.xml") as f2,
):
    GLYPHDATA = GlyphData.from_files(f1, f2)

non_spacing_marks_by_name = set()
non_spacing_marks_by_unicode = set()

for name, attrs in GLYPHDATA.names.items():
    category = attrs.get("category")
    sub_category = attrs.get("subCategory")
    if category == "Mark" and sub_category == "Nonspacing":
        non_spacing_marks_by_name.add(name)
        if "production" in attrs:
            non_spacing_marks_by_name.add(attrs["production"])
        if "altNames" in attrs:
            non_spacing_marks_by_name.update(attrs["altNames"].replace(" ", "").split(","))
        if "unicode" in attrs:
            non_spacing_marks_by_unicode.add(attrs["unicode"])
```

Reply:
Apparently openbinary is deprecated, will use the mechanism suggested by https://docs.python.org/3/library/importlib.resources.html#importlib.resources.open_binary

Reply:
Done

Reply:
you no longer need to check that the glyphdata_file is  at `glyphs_lib / "Lib/glyphsLib/data/GlyphData.xml"`, because you are now using importlib.resources to get to the data file, glyphsLib need only be importable now, doesn't matter where it is located on the filesystem.

Reply:
Oops, will fix.

Reply:
Hm. I think I already did.

Reply:
Ha, but only locally. Push is an important step.